File: snake-scala-2d/src/main/scala/metacake/scala/snake/input/KeyConfiguration.scala
package metacake.scala.snake.input

import java.awt.event.KeyEvent

import io.metacake.core.input.ActionTrigger
import io.metacake.core.process.ActionRecognizerName
import io.metacake.s2d.input.keyboard.KeyboardActionTrigger
import io.metacake.s2d.process.recognizers.keyboard.{KeyActionRecognizer, KeyClickActionRecognizer}

object KeyConfiguration {

  val W: ActionRecognizerName = new ActionRecognizerName("W")
  val wRecognizer: KeyActionRecognizer = new KeyClickActionRecognizer(W)
  val wTrigger: ActionTrigger[KeyEvent] = new KeyboardActionTrigger(KeyEvent.VK_W).bindRecognizer(wRecognizer)

  val S: ActionRecognizerName = new ActionRecognizerName("S")
  val sRecognizer: KeyActionRecognizer = new KeyClickActionRecognizer(S)
  val sTrigger: ActionTrigger[KeyEvent] = new KeyboardActionTrigger(KeyEvent.VK_S).bindRecognizer(sRecognizer)

  val A: ActionRecognizerName = new ActionRecognizerName("A")
  val aRecognizer: KeyActionRecognizer = new KeyClickActionRecognizer(A)
  val aTrigger: ActionTrigger[KeyEvent] = new KeyboardActionTrigger(KeyEvent.VK_A).bindRecognizer(aRecognizer)

  val D: ActionRecognizerName = new ActionRecognizerName("D")
  val dRecognizer: KeyActionRecognizer = new KeyClickActionRecognizer(D)
  val dTrigger: ActionTrigger[KeyEvent] = new KeyboardActionTrigger(KeyEvent.VK_D).bindRecognizer(dRecognizer)

  val ESCAPE: ActionRecognizerName = new ActionRecognizerName("ESC")
  val escRecognizer: KeyActionRecognizer = new KeyClickActionRecognizer(ESCAPE)
  val escTrigger: ActionTrigger[KeyEvent] = new KeyboardActionTrigger(KeyEvent.VK_ESCAPE).bindRecognizer(escRecognizer)
}

File: java-snake-2d/src/main/java/metacake/snake/input/KeyConfiguration.java
package metacake.snake.input;

import io.metacake.core.input.ActionTrigger;
import io.metacake.core.process.ActionRecognizerName;
import io.metacake.s2d.input.keyboard.KeyboardActionTrigger;
import io.metacake.s2d.process.recognizers.keyboard.KeyActionRecognizer;
import io.metacake.s2d.process.recognizers.keyboard.KeyClickActionRecognizer;

import java.awt.event.KeyEvent;

public class KeyConfiguration {
    public static final ActionRecognizerName W = new ActionRecognizerName();
    public static final KeyActionRecognizer wRecognizer = new KeyClickActionRecognizer(W);
    public static final ActionTrigger<KeyEvent> wTrigger = new KeyboardActionTrigger(KeyEvent.VK_W).bindRecognizer(wRecognizer);

    public static final ActionRecognizerName S = new ActionRecognizerName();
    public static final KeyActionRecognizer sRecognizer = new KeyClickActionRecognizer(S);
    public static final ActionTrigger<KeyEvent> sTrigger = new KeyboardActionTrigger(KeyEvent.VK_S).bindRecognizer(sRecognizer);

    public static final ActionRecognizerName A = new ActionRecognizerName();
    public static final KeyActionRecognizer aRecognizer = new KeyClickActionRecognizer(A);
    public static final ActionTrigger<KeyEvent> aTrigger = new KeyboardActionTrigger(KeyEvent.VK_A).bindRecognizer(aRecognizer);

    public static final ActionRecognizerName D = new ActionRecognizerName();
    public static final KeyActionRecognizer dRecognizer = new KeyClickActionRecognizer(D);
    public static final ActionTrigger<KeyEvent> dTrigger = new KeyboardActionTrigger(KeyEvent.VK_D).bindRecognizer(dRecognizer);
}